syntax = "proto3";

package yandex.mdb.cms.v1;
option go_package = "a.yandex-team.ru/cloud/mdb/cms/api/grpcapi/v1;cmsv1";

import "cloud/mdb/cms/api/grpcapi/v1/instance_operation.proto";

service InstanceService {
    rpc MoveInstance(MoveInstanceRequest) returns (InstanceOperation) {}
    rpc WhipPrimary(WhipPrimaryRequest) returns (InstanceOperation) {}
}

message MoveInstanceRequest {
    string instance_id = 1;
    string comment = 2; // ticket (or probably tickets) with reason
    bool force = 3;
}

message WhipPrimaryRequest {
    string instance_id = 1;
    string comment = 2; // ticket (or probably tickets) with reason
}
